Client/Frontend/Browser/Playlist/PlaylistSharedFolder.swift

```diff
@@ -0,0 +1,184 @@
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import Data
+import CoreData
+
+public struct PlaylistSharedFolderModel: Codable {
+  public let version: String
+  public let playlistId: String
+  public let folderName: String
+  public let folderImage: URL
+  public let creatorName: String
+  public let creatorLink: String
+  public let updateAt: String
+  public var mediaItems: [PlaylistInfo]
+  
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    version = try container.decode(String.self, forKey: .version)
+    playlistId = try container.decode(String.self, forKey: .playlistId)
+    folderName = try container.decode(String.self, forKey: .folderName)
+    folderImage = try container.decode(URL.self, forKey: .folderImage)
+    creatorName = try container.decode(String.self, forKey: .creatorName)
+    creatorLink = try container.decode(String.self, forKey: .creatorLink)
+    updateAt = try container.decode(String.self, forKey: .updateAt)
+    mediaItems = try container.decode([MediaItem].self, forKey: .mediaItems).map { item in
+      PlaylistInfo(name: item.title, src: item.url.absoluteString, pageSrc: item.url.absoluteString, pageTitle: item.title, mimeType: "video", duration: 0.0, detected: true, dateAdded: Date(), tagId: item.mediaItemId)
+    }
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(version, forKey: .version)
+    try container.encode(playlistId, forKey: .playlistId)
+    try container.encode(folderName, forKey: .folderName)
+    try container.encode(folderImage, forKey: .folderImage)
+    try container.encode(creatorName, forKey: .creatorName)
+    try container.encode(creatorLink, forKey: .creatorLink)
+    try container.encode(updateAt, forKey: .updateAt)
+    
+    let items = mediaItems.compactMap({ item -> MediaItem? in
+      guard let url = URL(string: item.pageSrc) else { return nil }
+      return MediaItem(mediaItemId: item.tagId.isEmpty ? UUID().uuidString : item.tagId,
+                title: item.pageTitle,
+                url: url)
+    })
+    
+    try container.encode(items, forKey: .mediaItems)
+  }
+  
+  public static func fetchPlaylist(playlistId: String) async throws -> PlaylistSharedFolderModel {
+    guard let playlistURL = URL(string: "https://brandon-t.github.io/playlist/static/\(playlistId == "SyncNow" ? "content" : "playlist").json?\(playlistId)") else {
+      throw "Invalid Playlist URL"
+    }
+    
+    let (data, _) = try await NetworkManager().dataRequest(with: playlistURL)
+    var item = try JSONDecoder().decode(Self.self, from: data)
+    item.mediaItems = await fetchMediaItemInfo(item: item)
+    return item
+  }
+  
+  @MainActor
+  public func createInMemoryStorage() async -> PlaylistFolder {
```

Comment:
This is kind of odd, why does this live on the model? It calls static methods?

Reply:
Moved out. Originally these were mutating and modifying the model directly to update an observer in SwiftUI.
Now they're not necessary as everything is back to being done in UIKit.